Lib/pathlib.py

```diff
@@ -707,6 +665,57 @@ def match(self, path_pattern, *, case_sensitive=None):
                 return False
         return True
 
+
+class PurePath(_BasePurePath):
+    """Base class for manipulating paths without I/O.
+
+    PurePath represents a filesystem path and offers operations which
+    don't imply any actual filesystem I/O.  Depending on your system,
+    instantiating a PurePath will return either a PurePosixPath or a
+    PureWindowsPath object.  You can also instantiate either of these classes
+    directly, regardless of your system.
+    """
+    __slots__ = ()
+
+    def __new__(cls, *args, **kwargs):
+        """Construct a PurePath from one or several strings and or existing
+        PurePath objects.  The strings and path objects are combined so as
+        to yield a canonicalized path, which is incorporated into the
+        new PurePath object.
+        """
+        if cls is PurePath:
+            cls = PureWindowsPath if os.name == 'nt' else PurePosixPath
+        return object.__new__(cls)
+
+    def __fspath__(self):
+        return str(self)
+
+    def __bytes__(self):
+        """Return the bytes representation of the path.  This is only
+        recommended to use under Unix."""
+        return os.fsencode(self)
+
+    def as_uri(self):
+        """Return the path as a 'file' URI."""
+        if not self.is_absolute():
+            raise ValueError("relative path can't be expressed as a file URI")
+
+        drive = self.drive
+        if len(drive) == 2 and drive[1] == ':':
+            # It's a path on a local drive => 'file:///c:/a/b'
+            prefix = 'file:///' + drive
+            path = self.as_posix()[2:]
+        elif drive:
+            # It's a path on a network drive => 'file://host/share/a/b'
+            prefix = 'file:'
+            path = self.as_posix()
+        else:
+            # It's a posix path => 'file:///etc/hosts'
+            prefix = 'file://'
+            path = str(self)
+        return prefix + urlquote_from_bytes(os.fsencode(path))
+
+
 # Can't subclass os.PathLike from PurePath and keep the constructor
 # optimizations in PurePath.__slots__.
```

Comment:
This comment is slightly strange: how can «constructor optimizations» be found in \_\_slots\_\_?

Reply:
This comment always puzzled me - it originally referred to a `_parse_args()` method, but that method never interfered with inheriting from `os.PathLike`.

I think the issue is that `os.PathLike` doesn't use `__slots__`, and so all subclasses have an instance dict. I may be wrong.

Reply:
I can't find any discussion of it in [PEP-519](https://peps.python.org/pep-0519/), so perhaps it's just an oversight. @brettcannon penny for your thoughts?

Reply:
@barneygale I take it you don't need that thought any more? (And FYI I live in Canada, so pennies are worthless to me, so you can keep your penny regardless. 😉)

Reply:
Turns out the comment was right all along. If we subclass `os.PathLike`, we incur the cost of the ABC `__instancecheck__`. Previously this occurred in `_parse_args()`, which is the function that was referred to in the comment before I changed it to `__slots__`. D'oh. 

PR that reverts the pathlib change: #105073